volatility3/framework/plugins/windows/driver_signing_enforcement.py

```diff
@@ -0,0 +1,124 @@
+
+vollog = logging.getLogger(__name__)
+
+"""
+A common technique by modern Windows rootkits is disabling Driver Signing Enforcement so that
+unsigned kernel rootkits can load. Rapid 7 documented many examples here:
+
+https://www.rapid7.com/blog/post/2021/12/13/driver-based-attacks-past-and-present/
+
+To disable DSE, malware will set the `g_CiOptions` global variable in the kernel to 0.
+
+This plugin locates this variable in memory and then reports its status and value.
+
+If this global variable is disabled on Windows 10 systems then it means the system is either in
+developer mode or that a rootkit purposely overwrote the value. Neither would be expected in production settings.
+"""
+class DriverSigningEnforcement(interfaces.plugins.PluginInterface):
+    """Reports the status of Driving Signing Enforcement"""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(name = 'kernel', description = 'Windows kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.PluginRequirement(name = 'modules', plugin = modules.Modules, version = (1, 0, 0)),
+            requirements.VersionRequirement(name = 'pdbutil', component = pdbutil.PDBUtility, version = (1, 0, 0)),
+        ]
+
+    def _get_ci_dll(self, kernel):
+        """
+        Finds the CI.DLL kernel module. This module is used for
+        Code Integrity enforcement, and is targeted by malware to disable security checks.
+
+        Args:
+            kernel: interfaces.context.ModuleInterface of the kernel
+        Returns:
+            The `_LDR_DATA_TABLE_ENTRY` of CI.DLL
+            None if module cannot be located
+        """
+        for mod in modules.Modules.list_modules(self.context, 
+                                                kernel.layer_name,
+                                                kernel.symbol_table_name):
+     
+            try:
+                module_name = mod.BaseDllName.get_string()
+            except exceptions.InvalidAddressException:
+                continue
+
+            if module_name.lower() == "ci.dll":
+                return mod
+
+        return None
+
+    def _check_win10_64bit(self, kernel):
+        """
+        Checks if the analyzed memory sample is from
+        a Windows 10 64bit system.
+
+        Returns:
+            bool
+        """
+        kuser = info.Info.get_kuser_structure(self.context, 
+                                              kernel.layer_name,
+                                              kernel.symbol_table_name)
+
+        nt_major_version = int(kuser.NtMajorVersion)
+
+        is_64bit = symbols.symbol_table_is_64bit(self.context, kernel.symbol_table_name)
+
+        return nt_major_version >= 6 and is_64bit
+    
+    def _generator(self):
```

Comment:
```suggestion
    def _generator(self)  -> Iterator[Tuple]:
```

Reply:
I'm not sure Iterator is imported, I'm all for adding the typing information, but it also needs to have the `from typing import Iterator` with it, if that's not already there.  That might be tricky to do in a suggestion, but the suggestion isn't that handy if it can't be automatically applied and needs more work on the file afterwards... 5:P